src/ophyd_async/core/_signal_backend.py

```diff
@@ -167,7 +167,7 @@ def _datakey_dtype_numpy(
 
 
 def _datakey_shape(value: SignalDatatype) -> list[int]:
-    if type(value) in _primitive_dtype or isinstance(value, StrictEnum):
+    if type(value) in _primitive_dtype or isinstance(value, EnumTypes):
```

Comment:
Does this work? isinstance expects a class or tuple of classes, here you're passing a Union of classes.

Same for the places above.

Reply:
```python
>>> class Thing():
...     pass
... 
>>> class Thing2():
...     pass
... 
>>> all_things = Thing | Thing2
>>> instance = Thing()
>>> isinstance(instance, all_things)
True

```

Looks like it does work for python 3.12 at least